mcs/class/System/Mono.AppleTls/AppleTlsContext.cs

```diff
@@ -341,10 +341,12 @@ void InitializeConnection ()
 				PeerDomainName = ServerName;
 			}
 
-			if (Options.AllowRenegotiation)
+			if (Options.AllowRenegotiation && Environment.OSVersion.Version >= MinRenegotiationVersion)
```

Comment:
will this work on iOS etc? I'm not sure which Environment.OSVersion we're getting there and the API is listed as iOS 10.0+, tvOS 10.0+, watchOS 3.0+.

shouldn't we just call CanRenegotiate? or is that just used on the server-side?

Reply:
I think we need better "checking" method which does include all MONOTOUCH profiles. The option is supported https://developer.apple.com/documentation/security/sslsessionoption/ksslsessionoptionallowrenegotiation?language=objc

XI does it like https://github.com/xamarin/xamarin-macios/blob/54101252aca7e15a9e4fd04e98d1bfb49e411f05/src/IOSurface/IOSurfacePropertyKey.cs#L34-L43